### DISCUSSION TITLE
How to resolve mergify backport conflicts

### DISCUSSION THREAD

**Discussion by wanpengfei-git** [Mergifyio/mergify#5057] [Q&A]
Since the backport process creates a separate branch, this results in many of our code contributors not having enough permissions to make changes based on conflicts in this branch's code after backporting to other branches, and also not having close permissions. Is there a perfect solution for this?
For example, mergify could provide a way for contributors with read access to use mergify's access to do push commits on this branch.

**Comment by sileht** [comment]
You can set the option: `bot_account: {{ author }}`, then the backport pull request will be created on behalf of the original PR author.
This doesn't solve the permissions issue, as the user should have permission to create branches and allow Mergify to act on his behalf (It just needs to login into the dashboard).

The permission issue can't be bypassed by Mergify in anyways, it's GitHub-controlled rules configurable with branch protection and Collaborator permissions settings.

**Reply by wanpengfei-git** [reply]
We don't want to give all contributors the write permissions, but mergify has create and push permissions, so my idea is that when there is a conflict, can we let contributors use mergify permissions to solve the conflict by merging code into the conflicting branch. Instead of having to submit the pr repeatedly without conflict and request the organization to close the conflicting pr and delete the conflicting branch, which would largely increase the complexity of cherry pick.

**Reply by jd** [reply]
> can we let contributors use mergify permissions to solve the conflict by merging code into the conflicting branch

No, you can't do that. There's no "Mergify context" on the branch.
GitHub just sees the branch as being a regular branch in your repository so non-collaborator don't have access.

The best way for them is to copy the branch to their fork, and open a new PR.

**Reply by wanpengfei-git** [reply]
Got it, thanks
